File: TravellerCreator/ClientApp/src/app/sector-creator/models/interfaces/star-system.ts
import {StarSystemType} from "../enums/star-system-type";
import {IStar} from "./star";
import {IPlanet} from "./planet";

export interface IStarSystem {
  stars: IStar[];
  type: StarSystemType;
  gasGiant: boolean;
  planets: IPlanet[];
}

File: TravellerCreatorModels/SectorCreator/Interfaces/StarSystem.cs
using TravellerCreatorModels.SectorCreator.Enums;

namespace TravellerCreatorModels.SectorCreator.Interfaces;

public interface IStarSystem
{
    public List<IStar> Stars { get; set; }
    public StarSystemType Type { get; set; }
    public bool GasGiant { get; set; }
    public List<IPlanet> Planets { get; set; }
}